perses/samplers/multistate.py

```diff
@@ -133,117 +51,87 @@ def setup(self, n_states, temperature, storage_file, minimisation_steps=100,
             difference = np.diff(lambda_schedule)
             assert ( all(i >= 0. for i in difference ) ), 'lambda_schedule must be monotonicly increasing'
 
+        #starting with the initial positions generated py geometry.py
         sampler_state =  SamplerState(positions, box_vectors=hybrid_system.getDefaultPeriodicBoxVectors())
-        sampler_state_list = []
         for lambda_val in lambda_schedule:
             compound_thermodynamic_state_copy = copy.deepcopy(compound_thermodynamic_state)
             compound_thermodynamic_state_copy.set_alchemical_parameters(lambda_val,lambda_protocol)
             thermodynamic_state_list.append(compound_thermodynamic_state_copy)
 
              # now generating a sampler_state for each thermodyanmic state, with relaxed positions
             context, context_integrator = context_cache.get_context(compound_thermodynamic_state_copy)
-            feptasks.minimize(compound_thermodynamic_state_copy,sampler_state) 
+            feptasks.minimize(compound_thermodynamic_state_copy,sampler_state)
             sampler_state_list.append(copy.deepcopy(sampler_state))
 
-         #nonalchemical_thermodynamic_states = [
-        #    ThermodynamicState(self._factory._old_system, temperature=temperature),
-        #    ThermodynamicState(self._factory._new_system, temperature=temperature)]
+        if endstates:
```

Comment:
perhaps this can go in a standalone function in `utils.py` since both neq and staged equilibrium methods need to be able to compute (m)BAR estimates w.r.t. endstates?